daemon/start.go

```diff
@@ -124,15 +124,16 @@ func (daemon *Daemon) containerStart(container *container.Container) (err error)
 		}
 	}
 
+	container.Unlock()
 	mounts, err := daemon.setupMounts(container)
 	if err != nil {
+		container.Lock()
 		return err
 	}
 	mounts = append(mounts, container.IpcMounts()...)
 	mounts = append(mounts, container.TmpfsMounts()...)
 
 	container.Command.Mounts = mounts
```

Comment:
Yes, I think @cpuguy83 is right, here we need a lock because container is modified.